File: app/src/main/java/com/example/mywechat/MainActivity.java
package com.example.mywechat;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setLoginView();
    }

    private void setLoginView() {
        setContentView(R.layout.activity_main);
        Button loginButton = findViewById(R.id.login_button);
        Button registerUserButton = findViewById(R.id.register_user_button);

        loginButton.setOnClickListener(v -> {
            //TODO: 增加登录验证
            startActivity(new Intent(MainActivity.this, UserActivity.class));
        });
        registerUserButton.setOnClickListener(v -> {
            setRegisterView();
        });
    }

    private void setRegisterView() {
        setContentView(R.layout.fragment_register);
        Button loginUserButton = findViewById(R.id.login_user_button);
        Button registerButton = findViewById(R.id.register_button);

        loginUserButton.setOnClickListener(v -> {
            setLoginView();
        });
        registerButton.setOnClickListener(v -> {
            // TODO: 进行注册验证
        });
    }

}

File: app/src/main/java/com/example/mywechat/MainActivity.kt
package com.example.mywechat;


import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.mywechat.ui.UserActivity
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun jumpToUser(username: String, password: String) {
        (application as App).username = username
        val intent: Intent = Intent(this, UserActivity::class.java)
        intent.putExtra("username", username)
        intent.putExtra("password", password)
        startActivity(intent)
        finish()
    }
}